scoreboard/log_import.py

```diff
@@ -1,157 +1,101 @@
-        # skip blank/junk lines
-        if not line.strip():
-            continue
-        yield LogfileLine(line, logfile.src)
-        if lines % 10000 == 0:
-            model.save_logfile_progress(s, logfile.path, f.tell())
-    model.save_logfile_progress(s, logfile.path, f.tell())
-    end = time.time()
-    msg = "Finished reading {f} ({l} new lines) in {s} secs"
-    print(msg.format(f=logfile.path, l=lines, s=round(end - start, 2)))
+    url = api_url
+    current_key = model.get_logfile_progress(s, url).current_key
 
+    while True:
+        r = request_logfile_lines(url, current_key)
+        try:
+            response = r.json()
+        except Exception:
+            print("Failed to decode into json")
+            print(r.text)
+            raise
+        assert response['status'] == 200 and response['message'] == 'OK'
 
-def parse_logfile_line(s: sqlalchemy.orm.session.Session,
-                       line: LogfileLine) -> Optional[dict]:
-    """Read a single logfile line and insert it into the database.
+        if not len(response['results']):
+            break
 
-    If the game is not valid, None is returned. Invalid games could be:
-        * a non-vanilla crawl game (eg sprint or zotdef)
-        * Corrupt data
+        for game in response['results']:
+            add_game(s, game)
+            games += 1
+            if games % 10000 == 0:
+                print("Processed %s games..." % games)
 
-    """
-    # Early warning of corruption
-    if '\x00' in line.data:
-        return None
+        current_key = response['next_offset']
+        model.save_logfile_progress(s, url, current_key)
+        s.commit()
+    s.commit()
+    end = time.time()
+    print("Loaded %s new games in %s secs" % (games, round(end - start, 2)))
 
-    game = {'src': line.src}
 
-    # Parse the log's raw field data
-    for field in re.split(LINE_SPLIT_PATTERN, line.data):
-        if not field.strip():
-            continue
-        try:
-            k, v = parse_field(field)
-        except FieldParseError as e:
-            print("Can't parse field {}: {}, skipping line: {}".format(
-                field, e, line))
-            return None
-        game[k] = v
+def add_game(s: sqlalchemy.orm.session.Session, api_game: dict) -> bool:
+    """Add a game to the database.
 
+    Returns True if a game was found and successfully added.
+    """
     # Validate the data -- some old broken games don't have this field and
     # and should be ignored.
-    if 'start' not in game:
+    if 'start' not in api_game['data']:
         return None
-    if 'v' not in game:
-        print("Found game without version tag, skipping. Line: %s, Game: %s" %
-              (line.data, game))
+    if 'v' not in api_game['data']:
         return None
     # We should only parse vanilla dcss games
-    if game['lv'] != '0.1':
+    if api_game['data']['lv'] != '0.1':
         return None
-    game['gid'] = calculate_game_gid(game)
+
+    game = {}
```

Comment:
I create a `game` dict here, then later I create `gamedict`. This can be simplified. Or at least add an `TODO` as a warning for the future.